canonical-web-and-design/snapcraft.io
# Check if snapcraft.yaml is present

## Done

In this PR we are using the GitHub GraphQL API to:

- List all the user repositories and mark the ones with a snapcraft.yaml present:
![github-graphql1](https://user-images.githubusercontent.com/6353928/73292414-ee5dac80-41f9-11ea-9998-682ccaedc31a.png)


- Show an error on the builds page when the selected repo for the snap doesnt contain a snapcraft.yaml:
![github-graphql2](https://user-images.githubusercontent.com/6353928/73292597-38df2900-41fa-11ea-9265-cf2035f6c331.png)


## Issue / Card

Fixes #2449

## QA

- Pull the branch
- Run the site using the command `./run`
- View the site locally in your web browser at: http://0.0.0.0:8004/
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Login as a publisher
- Go to http://0.0.0.0:8004/<your_snap>/builds and check to you can:
- Login with GitHub
- Try to link you snap with a github repo and check that the repos contains a mark when snapcraft.yaml is present
- Check if a snap with a github repo without snapcraft.yaml get the error
